benchmarks/benchmark_dataset.py

```diff
@@ -815,3 +828,103 @@ def sample(self,
                 ))
         self.maybe_oversample_requests(sampled_requests, num_requests)
         return sampled_requests
+
+
+# -----------------------------------------------------------------------------
+# Next Edit Prediction Dataset Implementation
+# -----------------------------------------------------------------------------
+zeta_prompt = """### Instruction:
+You are a code completion assistant and your task is to analyze user edits and then rewrite an excerpt that the user provides, suggesting the appropriate edits within the excerpt, taking into account the cursor location.
+
+### User Edits:
+
+{}
+
+### User Excerpt:
+
+{}
+
+### Response:
+
+""" # noqa: E501
+
+
+def _format_zeta_prompt(
+        examples: dict,
+        original_start_marker: str = "<|editable_region_start|>") -> dict:
+    """Format the zeta prompt for the Next Edit Prediction (NEP) dataset.
+    
+    This function formats examples from the NEP dataset 
+    into prompts and expected outputs. It could be 
+    further extended to support more NEP datasets.
+    
+    Args:
+        examples: The dataset examples containing events, 
+            inputs, and outputs.
+        original_start_marker: The marker indicating the 
+            start of the editable region. Defaults to 
+            "<|editable_region_start|>".
+            
+    Returns:
+        A dictionary with the formatted prompts and expected outputs.
+    """
+    events = examples["events"]
+    inputs = examples["input"]
+    outputs = examples["output"]
+    prompts, expected_outputs = [], []
+    for event, input, output in zip(events, inputs, outputs):
+        # create the final prompt
+        prompts.append(zeta_prompt.format(event, input))
+
+        # following the original implementation, extract the focused region
+        # from the raw output
+        output_start_index = output.find(original_start_marker)
+        output_focused_region = output[output_start_index:]
+        output = output_focused_region
+        expected_outputs.append(output)
+
+    return {"prompt": prompts, "expected_output": expected_outputs}
+
+
+class NextEditPredictionDataset(HuggingFaceDataset):
+    """
+    Dataset class for processing a Next Edit Prediction dataset.
+    """
+
+    SUPPORTED_DATASET_PATHS = {
+        "zed-industries/zeta",
+    }
+    MAPPING_PROMPT_FUNCS = {
+        "zed-industries/zeta": _format_zeta_prompt,
+    }
+
+    def load_data(self) -> None:
+        """Load data from HuggingFace datasets."""
+        super().load_data()
+        formatting_prompt_func = self.MAPPING_PROMPT_FUNCS.get(
+            self.dataset_path)
+        if formatting_prompt_func is None:
+            raise ValueError(f"Unsupported dataset path: {self.dataset_path}")
+        self.data = self.data.map(formatting_prompt_func, batched=True)
```

Comment:
have you considered formatting "lazily" only the `num_requests` samples you get in sample?